backend/danswer/db/document_set.py

```diff
@@ -0,0 +1,282 @@
+        db_session.commit()
+    except:
+        db_session.rollback()
+        raise
+
+
+def fetch_document_sets(
+    db_session: Session,
+) -> list[tuple[DocumentSetDBModel, list[ConnectorCredentialPair]]]:
+    """Return is a list where each element contains a tuple of:
+    1. The document set itself
+    2. All CC pairs associated with the document set"""
+    results = cast(
+        list[tuple[DocumentSetDBModel, ConnectorCredentialPair]],
+        db_session.execute(
+            select(DocumentSetDBModel, ConnectorCredentialPair)
+            .join(
+                DocumentSet__ConnectorCredentialPair,
+                DocumentSetDBModel.id
+                == DocumentSet__ConnectorCredentialPair.document_set_id,
+            )
+            .join(
+                ConnectorCredentialPair,
+                ConnectorCredentialPair.id
+                == DocumentSet__ConnectorCredentialPair.connector_credential_pair_id,
+            )
+            .where(
+                DocumentSet__ConnectorCredentialPair.is_current == True  # noqa: E712
+            )
+        ).all(),
+    )
+
+    aggregated_results: dict[
+        int, tuple[DocumentSetDBModel, list[ConnectorCredentialPair]]
+    ] = {}
+    for document_set, cc_pair in results:
+        if document_set.id not in aggregated_results:
+            aggregated_results[document_set.id] = (document_set, [cc_pair])
+        else:
+            aggregated_results[document_set.id][1].append(cc_pair)
+
+    return [
+        (document_set, cc_pairs)
+        for document_set, cc_pairs in aggregated_results.values()
+    ]
+
+
+def fetch_documents_for_document_set(
+    document_set_id: int, db_session: Session, current_only: bool = True
+) -> Sequence[Document]:
+    stmt = (
+        select(Document)
+        .join(
+            DocumentByConnectorCredentialPair,
+            DocumentByConnectorCredentialPair.id == Document.id,
+        )
+        .join(
+            ConnectorCredentialPair,
+            and_(
+                ConnectorCredentialPair.connector_id
+                == DocumentByConnectorCredentialPair.connector_id,
+                ConnectorCredentialPair.credential_id
+                == DocumentByConnectorCredentialPair.credential_id,
+            ),
+        )
+        .join(
+            DocumentSet__ConnectorCredentialPair,
+            DocumentSet__ConnectorCredentialPair.connector_credential_pair_id
+            == ConnectorCredentialPair.id,
+        )
+        .join(
+            DocumentSetDBModel,
+            DocumentSetDBModel.id
+            == DocumentSet__ConnectorCredentialPair.document_set_id,
+        )
+        .where(DocumentSetDBModel.id == document_set_id)
+    )
+    if current_only:
+        stmt = stmt.where(
+            DocumentSet__ConnectorCredentialPair.is_current == True  # noqa: E712
+        )
+    stmt = stmt.distinct()
+
+    return db_session.scalars(stmt).all()
+
+
+def fetch_document_sets_for_documents(
```

Comment:
Let's discuss how we can specify user access to document sets, right now everything is fully open and filtered at the Vespa level by ACL right?